src/plugins/discover/public/application/main/utils/fetch_all.ts

```diff
@@ -131,6 +132,70 @@ export function fetchAll(
   }
 }
 
+export function fetchMoreDocuments(
+  dataSubjects: SavedSearchData,
+  fetchDeps: FetchDeps
+): Promise<void> {
+  const { getAppState, services, savedSearch } = fetchDeps;
+  const searchSource = savedSearch.searchSource.createChild();
+
+  try {
+    const lastDocuments = dataSubjects.documents$.getValue().result || [];
+    const lastDocumentSort = lastDocuments[lastDocuments.length - 1]?.raw?.sort;
+
+    if (!lastDocumentSort) {
+      return Promise.resolve();
+    }
+
+    searchSource.setField('searchAfter', lastDocumentSort);
+
+    const dataView = searchSource.getField('index')!;
+    const query = getAppState().query;
+    const recordRawType = getRawRecordType(query);
+
+    if (recordRawType === RecordRawType.PLAIN) {
+      // not supported yet
+      // TODO?
+      return Promise.resolve();
+    }
+
+    // Update the base searchSource, base for all child fetches
+    updateVolatileSearchSource(searchSource, {
+      dataView,
+      services,
+      sort: getAppState().sort as SortOrder[],
+    });
+
+    // Mark subjects as loading
+    sendLoadingMoreMsg(dataSubjects.documents$);
+
+    // Start fetching all required requests
+    const response = fetchDocuments(searchSource, fetchDeps);
+
+    // Handle results of the individual queries and forward the results to the corresponding dataSubjects
+    response
+      .then(({ records, textBasedQueryColumns }) => {
+        dataSubjects.documents$.next({
+          fetchStatus: FetchStatus.COMPLETE,
+          result: [...(dataSubjects.documents$.getValue().result || []), ...records],
+          textBasedQueryColumns,
+          recordRawType,
+          query,
+        });
+      })
+      .catch((error) => {
+        // TODO?
+        // eslint-disable-next-line no-console
+        console.error(error);
+      });
```

Comment:
Ideally yes, some sort of error indication would be great, currently users just get an infinite spinning link when it doesn't work

Reply:
tested, and there's an error toast, thx and resolving conversation